### PR TITLE
Getting generated MetaItem's as an IItemStack

### PR DESCRIPTION
**Pull request by idcppl**
**What:**  
This would allow you to get material generated IItemStack's through the bracket handler, and would make looping over MetaItem's everytime not a thing. Also somewhat related to #1335 
So the bracket handler could also accept the oreDict name, for example, `<metaitem:plateIron>`

### CODE REVIEW

File: src/main/java/gregtech/common/CommonProxy.java [hunk 1]
```diff
@@ -173,6 +174,7 @@ public static void registerRecipesLowest(RegistryEvent.Register<IRecipe> event)
         DecompositionRecipeHandler.runRecipeGeneration();
         RecyclingRecipes.init();
         WoodMachineRecipes.init();
+        MetaItemBracketHandler.rebuildComponentRegistry();
```
Comment (LAGIdiot) [inline src/main/java/gregtech/common/CommonProxy.java:177]: Based on what did you choose this place?

**Review by LAGIdiot — COMMENTED** [review]
Can you please be more descriptive in PR opening comment and provide some real use cases.
As I am not that familiar with this matter provided information does not say much to me.

File: src/main/java/gregtech/common/CommonProxy.java (hunk 1 above)
Comment (idcppl) [reply src/main/java/gregtech/common/CommonProxy.java:177]: This is the same registry spot that CraftTweaker uses for their bracket handlers.

**Comment by idcppl** [comment]
This would allow for all Metaitems that gregtech generates to be gotten in an IItemStack form that doesn't have to rely on getting the first in oredict. There is probably some more things I could make smoother. If you really don't care for the material based generation items being in this, I could just remove that part. The old system looped everytime the bracket was called and very inefficient.

**Comment by idcppl** [comment]
An example would be,  
```zs
RecipeMap.getByName("macerator").recipeBuilder()
.inputs([<ore:ingotSteel>])
.outputs(<metaitem:ingotSteel>)
.duration(60)
.EUt(30)
.buildAndRegister();

vs

RecipeMap.getByName("macerator").recipeBuilder()
.inputs([<ore:ingotSteel>])
.outputs(<ore:ingotSteel>.firstItem)
.duration(60)
.EUt(30)
.buildAndRegister();
```
It's almost the same, but the metaitem bracket handler guarantee's that it'll be GTCE ingot, only if it's generated by GTCE, else just use `<ore:ingotSteel>.firstItem`

File: src/main/java/gregtech/api/recipes/crafttweaker/MetaItemBracketHandler.java [hunk 2]
```diff
@@ -6,34 +6,58 @@
 import crafttweaker.api.item.IItemStack;
 import crafttweaker.mc1120.item.MCItemStack;
 import crafttweaker.zenscript.IBracketHandler;
+import gregtech.api.items.materialitem.MaterialMetaItem;
 import gregtech.api.items.metaitem.MetaItem;
 import gregtech.api.items.metaitem.MetaItem.MetaValueItem;
+import gregtech.api.unification.OreDictUnifier;
+import gregtech.common.items.MetaItems;
```
Comment (LAGIdiot) [inline src/main/java/gregtech/api/recipes/crafttweaker/MetaItemBracketHandler.java:13]: This coupling is wrong. We should not access GTCE common from API. Please look for other way how to do it.

File: src/main/java/gregtech/common/CommonProxy.java (hunk 1 above)
Comment (LAGIdiot) [reply src/main/java/gregtech/common/CommonProxy.java:177]: That seems like good place then.

**Review by LAGIdiot — CHANGES_REQUESTED** [review]
Good work updating PR. There is one thing that need to be resolved everything else seems reasonable.

**Comment by idcppl** [comment]
Believe this should be finished, now that common is out of api area.

**Review by LAGIdiot — APPROVED** [review]
Thank you for finishing this up.

**Pull request merged**